### DISCUSSION TITLE
Can you additional more send/receive feature?

### DISCUSSION THREAD

**Discussion by rieniter** [jchristn/SuperSimpleTcp#96] [Ideas]
I suggest add more argument in sending/receiving feature, such as:
Send( ipPort, "key_a",data)
Receive("key_a", data)

key_a is an key string, so the client and server can have more multiple sending/receiving for easy manage

Currently, it's only simple  Send(ipPort, data) and only 1 DataReceived event each client/server

**Comment by jchristn** [comment]
Hi @rieniter it's intentional that the library not do that.  The primary reason is that this library does not provide any form of framing, which would be necessary to indicate to the other side what the value of the ```key``` was in your example. 

If you need this kind of functionality I would recommend either A) creating your own data structure to wrap messages that indicates the type of message or a message identifier, or B) take a look at [WatsonTcp](https://github.com/jchristn/watsontcp), which provides integrated framing.
